core/example/solvers/main.cpp

```diff
@@ -0,0 +1,158 @@
+#include "definitions.h"
+#include "particles.h"
+//#include "ewald.h"
+#include "direct.h"
+#include "pme.h"
+//#include "mesh.h"
+
+#include <iomanip>
+#ifdef TDS_BENCHMARKING
+#include <fstream>
+#endif
+
+#define CRYSTAL_SIZE 2
+
+int main(int argc, char** argv)
+{
+  // Initialize the kokkos runtime.
+  Kokkos::initialize( argc, argv );
+
+
+#ifdef TDS_BENCHMARKING
+  std::ofstream directfile;
+  std::ofstream ewaldfile;
+  std::ofstream pmefile;
+#endif
+
+  // crystal size
+  int c_size = (argc >= 2)?atoi(argv[1]):CRYSTAL_SIZE;
+  // number of periodic shells (and k_max for Ewald)
+  int periodic_shells = (argc >= 3)?atoi(argv[2]):0;
+  // accuracy parameter for the tuning of Ewald
+  double accuracy = (argc >= 4)?atof(argv[3]):1e-6;
+  // flag to set for the activation of solvers:
+  // 0 -> none
+  // 1 -> direct
+  // 2 -> Ewald
+  // 3 -> both
+  int flag = (argc >= 5)?atoi(argv[4]):3;
+  double l[3];
+  l[0] = l[1] = l[2] = 0.5*(double)c_size;
+  // alpha splitting parameter for Ewald
+  double alpha = (argc >= 6)?atof(argv[5]):2.0;
+  // cutoff radius for real-space part of Ewald
+  double r_max = (argc >= 7)?atof(argv[6]):0.499*(double)c_size;
+
+  int n_particles = c_size * c_size * c_size;
+  int n_meshpoints = 16*16*16;//Arbitrary value for now...
+ 
+#ifndef TDS_BENCHMARKING
+#else
+  directfile.open("direct.res", std::ofstream::out | std::ofstream::app);
+  ewaldfile.open("ewald.res", std::ofstream::out | std::ofstream::app);
+  pmefile.open("pme.res", std::ofstream::out | std::ofstream::app);
+#endif
+
+  ParticleList* particles = new ParticleList( n_particles ); 
+  //ParticleList particles = *plist;
+
+  ParticleList* mesh = new ParticleList( n_meshpoints );
+
+  std::cout << std::setprecision(12);
+
+/*  if (flag & 1)
```

Comment:
what is this commented section of code for? Looks similar to the section below

Reply:
I am restoring this currently. The original version allowed for a solver flag to be specified on the command line. A user could then select a flag to run the direct sum, ewald sum, spme, or all 3.